src/response_flags.rs

```diff
@@ -0,0 +1,264 @@
+        let mut last_access: Option<u32> = None;
+        let mut ttl: Option<i32> = None;
+        let mut client_flag: Option<u32> = None;
+        let mut win: Option<bool> = None;
+        let mut stale: bool = false;
+        let mut size: Option<u32> = None;
+        let mut opaque: Option<Vec<u8>> = None;
+
+        let mut n = start;
+        while n < header.len() {
+            let flag = header[n];
+            // Move past the flag
+            n += 1;
+            match flag {
+                b' ' => {
+                    // Space, skip it
+                    continue;
+                }
+                b'c' => {
+                    // cas_token flag (u32)
+                    let (v, len) = u32::from_radix_10(&header[n..]);
+                    if len > 0 {
+                        cas_token = Some(v);
+                        n += len;
+                    } else {
+                        n = find_space_or_end(&header, n);
+                    }
+                }
+                b'h' => {
+                    // fetched flag (bool) encoded as 1 or 0
+                    match header[n] {
+                        b'1' => {
+                            fetched = Some(true);
+                        }
+                        b'0' => {
+                            fetched = Some(false);
+                        }
+                        _ => {}
+                    }
+                    n += 1;
+                }
+                b'l' => {
+                    // last_access flag (u32)
+                    let (v, len) = u32::from_radix_10(&header[n..]);
+                    if len > 0 {
+                        last_access = Some(v);
+                        n += len;
+                    } else {
+                        n = find_space_or_end(&header, n);
+                    }
+                }
+                b't' => {
+                    // ttl flag (i32) encoded as -1 (for no ttl) or a positive number
+                    if header[n + 1] == b'-' {
+                        ttl = Some(-1);
+                        n += 2;
+                    } else {
+                        let (v, len) = i32::from_radix_10(&header[n..]);
+                        if len > 0 {
+                            ttl = Some(v);
+                            n += len;
+                        } else {
+                            n = find_space_or_end(&header, n);
+                        }
+                    }
+                }
+                b'f' => {
+                    // client_flag flag (u32)
+                    let (v, len) = u32::from_radix_10(&header[n..]);
+                    if len > 0 {
+                        client_flag = Some(v);
+                        n += len;
+                    } else {
+                        n = find_space_or_end(&header, n);
+                    }
+                }
+                b'W' => {
+                    // win flag (bool), no value
+                    win = Some(true);
```

Comment:
Just thinking about validation and exception safety with this parser.
* Is it worth exploding & failing if mutually exclusive flags like `W` and `Z` arrive in the same command?
* What happens if flags like `size` aren't properly followed by a parseable base10 `u32`? Does it crash the process?
* Are there other strings where the protocol allows which this parser doesn't, eg. `t-999` above?

Reply:
While the protocol is clear and will not put bad things on the wire, we should try to make it robust and never panic, in case the socket is desynchronized for some reason or a cosmic ray flips a bit. I have an update using from_radix_10_checked that does check ranges and others.